bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -496,33 +594,60 @@ public void actionPerformed(final ActionEvent e) {
 
     @SuppressWarnings("FieldCanBeLocal")
     private final ActionListener removeListener = evt -> {
-        TargetEnvironment env = getDataObject().getTargetEnvironment();
-        if (env.isBasePosition(_curPos)) {
-            DialogUtil.error("You can't remove the Base Position.");
-        } else if (_curPos != null) {
-            env = env.removeTarget(_curPos);
-        } else if (_curGroup != null) {
-            final GuideGroup primary = env.getOrCreatePrimaryGuideGroup();
-            if (_curGroup == primary) {
-                DialogUtil.error("You can't remove the primary guide group.");
-                return;
-            }
-            env = env.removeGroup(_curGroup._1());
-            final int groupIndex = env.getGuideEnvironment().getPrimaryIndex();
-            final GuideGroup group = env.getGuideEnvironment().getPrimary();
-            _curGroup = new Pair<>(groupIndex, group);
-        }
-        getDataObject().setTargetEnvironment(env);
-        final SPTarget selTarget = TargetSelection.get(env, getNode());
-        if (env.getTargets().contains(selTarget)) {
-            if (_curPos != null) _curPos.deleteWatcher(posWatcher);
-            _curPos = selTarget;
-            if (_curPos != null) {
-                _curPos.addWatcher(posWatcher);
-                refreshAll();
-                updateRemovePrimaryButtonsAndDetailEditor(env);
-            }
-        }
+        final TargetEnvironment envOld = getDataObject().getTargetEnvironment();
+
+        // If there is a selection, then handle it, separately for targets and guide groups.
+        final Option<TargetEnvironment> envNewOpt = _curSelection.flatMap(either -> either.fold(
+                // Handle targets.
+                t -> {
+                    if (selectionIsBasePosition()) {
+                        DialogUtil.error("You can't remove the base position.");
+                    } else if (selectionIsAutoTarget()) {
+                        DialogUtil.error("You can't remove automatic guide stars.");
+                    } else {
+                        return new Some<>(envOld.removeTarget(t));
```

Comment:
In theory this could be problematic because `removeTarget` will remove all instances of the target from the environment.  Is there a way to just remove the selected instance?